nuage-tripleo-heat-templates/docker/services/nova-compute-avrs.yaml

```diff
@@ -306,5 +306,41 @@ outputs:
 
       upgrade_tasks:
         get_attr: [NovaComputeAvrsDockerBase, role_data, upgrade_tasks]
+      update_tasks:
+        list_concat:
+        - {get_attr: [NovaComputeAvrsDockerBase, role_data, update_tasks]}
+        - - name: Is virtual-accelerator-base going to be updated
+            shell: yum check-update virtual-accelerator-base
+            register: virtual_accelerator_base_update_check
+            failed_when: virtual_accelerator_base_update_check.rc not in [0, 100]
+            changed_when: virtual_accelerator_base_update_check.rc == 100
+            when: step|int == 1
+          - name: Set virtual_accelerator_base_needs_update fact
+            set_fact: virtual_accelerator_base_needs_update={{ virtual_accelerator_base_update_check.rc == 100 }}
+            when: step|int == 1
+          - name: Stop VA on AVRS nodes before AVRS upgrade
+            service:
+              name: virtual-accelerator
+              state: stopped
+            when: (step|int == 1) and (virtual_accelerator_base_needs_update)
+          - name: Start Virtual Accelerator
+            systemd:
+              state: started
+              enabled: yes
+              daemon_reload: yes
+              name: virtual-accelerator
+            when: (step|int == 4) and (virtual_accelerator_base_needs_update)
+          - name: Restart network service on the AVRS nodes
+            service:
+              name: network
+              state: restarted
+            async: 2
+            poll: 0
+            when: (step|int == 4) and (virtual_accelerator_base_needs_update)
```

Comment:
I'm not sure virtual_accelerator_base_needs_update will be set correctly due to it only being registered in step 1.  There's actually an ansible issue where register is rerun even if there's a when condition that would be false, so it's likely that this wouldn't actually get set correctly in step 4.  You'd likely want to use some other mechanism (like touch a file in /tmp/, restart and remove tmp file) between step1/step4 to issue the restart.